airbyte-integrations/connectors/source-google-search-console/source_google_search_console/streams.py

```diff
@@ -376,23 +376,31 @@ class SearchByKeyword(SearchAnalytics):
     https://developers.google.com/webmaster-tools/v1/how-tos/all-your-data#search-appearance-data
     """
 
+    def stream_slices(
+        self, sync_mode: SyncMode, cursor_field: List[str] = None, stream_state: Mapping[str, Any] = None
+    ) -> Iterable[Optional[Mapping[str, Any]]]:
+        search_appearance_stream = SearchAppearance(self._session.auth, self._site_urls, self._start_date, self._end_date)
+
+        for stream_slice in super().stream_slices(sync_mode, cursor_field, stream_state):
+            keywords_records = search_appearance_stream.read_records(
+                sync_mode=SyncMode.full_refresh, stream_state=stream_state, stream_slice=stream_slice
+            )
+            keywords = {record["searchAppearance"] for record in keywords_records}
+
+            for keyword in keywords:
+                filters = {"dimension": "searchAppearance", "operator": "equals", "expression": keyword}
+                stream_slice["dimensionFilterGroups"] = [{"groupType": "and", "filters": filters}]
```

Comment:
Can we add a comment about that "or" is not supported? Just in case any refactor ninja 🥷.

Reply:
added comment